### DISCUSSION TITLE
Is There a Proper Way to Set Up a Custom Role for Admin Dashboard?

### DISCUSSION THREAD

**Discussion by ahmersaud** [supabase/supabase#29608] [Questions]
Hello everyone,

I'm working on setting up an admin panel for my Supabase PostgreSQL database, and I've come across various approaches. Some solutions suggest using custom claims, while others recommend creating custom roles. The range of methods is as diverse as the colors in a rainbow, but I haven't found a definitive solution yet.

What I need is to create a custom role, let's call it "admin," that bypasses all the security rules on the public schema and can log in. Afterward, I want to assign this role to one of the users who already has the "authenticated" role.

I tried doing this using the SQL editor. I created the role, granted all the necessary privileges, and assigned it to a user. However, when I logged in with that user, the client-side threw an error saying it couldn't assign the admin role.

What am I missing? Has anyone faced a similar issue, or is there a specific method I should follow to create and assign this custom role correctly

**Comment by silentworks** [comment]
You are asking about two different things here, one is a database role (custom role) and the other is an application role (custom claims). You should decide which you are trying to accomplish before moving on to the next step. Database roles should only be used when you fully understand how they work and the implications of using them, you can assign multiple database roles to a database user, do note database user and application users are two different things.

**Reply by ahmersaud** [reply]
Thanks for your reply! I understand the distinction you’re making between database roles and application roles. However, I’m not asking about two different things here. I’m simply looking for a solution where I can create an "admin" role in my Supabase PostgreSQL database that bypasses all security rules for the public schema. I want to assign this role to one of my users who already has the "authenticated" role, specifically for use within an admin panel I'm building, where the admin user has full access to everything.

Any guidance on achieving this setup would be greatly appreciated!

**Comment by marcusabaker35** [comment]
That Figma plugin for AI-powered conversion to code is a game-changer for design-to-development workflows, especially when building dashboards or admin interfaces. It aligns well with tools like the Free React Admin Theme that support Supabase integration, real-time database, and clean UI components. You can see it here: https://mobisoftinfotech.com/tools/free-react-admin-template-pills-of-zen-theme-docs/ . What part of the design-to-code process do you think benefits most from automation in dashboard development?
